raiden/network/transport/matrix/client.py

```diff
@@ -541,25 +531,9 @@ def _handle_response(self, response: Dict[str, Any], first_sync: bool = False) -
                     if should_call:
                         listener["callback"](event)
 
-            for event in sync_room["account_data"]["events"]:
-                room.account_data[event["type"]] = event["content"]
-
-        if first_sync:
```

Comment:
please, also remove the flag from the function signature.

Reply:
We still need that unfortunately: https://github.com/raiden-network/raiden/blob/535946c09fdd1a69c68feca78eb72b9c0874247b/raiden/network/transport/matrix/client.py#L455

Reply:
that can go away too, it is done as part of the `listen_forever` thread lifecycle:

https://github.com/raiden-network/raiden/blob/535946c09fdd1a69c68feca78eb72b9c0874247b/raiden/network/transport/matrix/client.py#L265

Reply:
No, the first sync is (and has to be) performed before the listen thread is started. That's why the check is there.

Reply:
I know ... what is the problem? Just remove the `if` here and the `if` on the top of the function.

Reply:
The reason the condition in (what is now) `_handle_response()` is there is because without it we had cases (esp. in the tests) where processing the remaining messages would delay transport shutdown long enough to cause problems (e.g. test timeouts)